### PR TITLE
Added useLocalStorage hook

### LINKED ISSUE #2: Create useLocalStorage hook
# Create `useLocalStorage` hook
The hook behaves similar to a `useState` hook, except the data is synchronized with the Local Storage API and persists within the browser's storage.

If approved and assigned, I would like to work on it as a contributor in Hacktoberfest.

### PR DESCRIPTION
**Pull request by paramsiddharth**
# Added `useLocalStorage` hook
Created a file for the `useLocalStorage` hook.
Resolves #2 .

[Edit xenodochial-chatelet-ww02b](https://codesandbox.io/s/xenodochial-chatelet-ww02b?fontsize=14&hidenavigation=1&theme=dark)

### CODE REVIEW

File: src/hooks/useLocalStorage/useLocalStorage.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { useState } from "react";
+
+// Hook
```
Comment (golu360) [inline src/hooks/useLocalStorage/useLocalStorage.js:3]: Remove all comments from this please

File: src/hooks/useLocalStorage/useLocalStorage.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+function useLocalStorage(key, initialValue) {
```
Comment (golu360) [inline src/hooks/useLocalStorage/useLocalStorage.js:4]: Please prefer ES6 Function Syntax

File: src/hooks/useLocalStorage/useLocalStorage.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      // Save to local storage
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (error) {
+      // A more advanced implementation would handle the error case
+      console.log(error);
+    }
+  };
+  return [storedValue, setValue];
+}
+
```
Comment (golu360) [inline src/hooks/useLocalStorage/useLocalStorage.js:37]: Please include a working demonstration of codesandbox.io of the same. That would be helpful to demonstrate exact usage of this hook.

**Review by golu360 — CHANGES_REQUESTED** [review]

**Comment by paramsiddharth** [comment]
@golu360 Finished updating the file. The CodeSandbox example is added in the PR description. Should I add it as a comment to the file too?

File: src/hooks/useLocalStorage/useLocalStorage.js (hunk 3 above)
Comment (golu360) [reply src/hooks/useLocalStorage/useLocalStorage.js:26]: Hi, for this, could you seperate your hook and main App.js into two different files? Make a file named useLocalStorage.js and import it's default export in the App.js to demonstrate it.

Check existing codesandboxes for a reference 😄

**Review by golu360 — CHANGES_REQUESTED** [review]

**Comment by golu360** [comment]
> @golu360 Finished updating the file. The CodeSandbox example is added in the PR description. Should I add it as a comment to the file too?

No, that is fine. We can update that in the README. I'll handle that

**Comment by golu360** [comment]
Looks good. Merging it.

Cheers 🥂

**Review by golu360 — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@golu360 Done! I added it to the README too. Please review and verify.

**Review by golu360 — APPROVED** [review]

**Pull request merged**